### PR TITLE
[ROCm] add Conv, NhwcConv benchmark to microbench

### PR DESCRIPTION
**Pull request by PeixuanZuo**
Add Conv, NhwcConv benchmark to microbench.

Related PR: https://github.com/microsoft/onnxruntime/pull/14982, https://github.com/microsoft/onnxruntime/pull/14980

### CODE REVIEW

File: onnxruntime/python/tools/microbench/conv.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    w: int
+
+    data_type: type
+
+
+class BenchmarkConv(BenchmarkOp):
+    def __init__(self, args):
+        BenchmarkOp.__init__(self, args)
+
+    def create_inputs_outputs(cls, op_param):
```
Comment (mindest) [inline onnxruntime/python/tools/microbench/conv.py:28]: Please add `@classmethod` decorator here.

File: onnxruntime/python/tools/microbench/conv.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    def create_cases(self):
+        # attributes of model : kernel_shape(3,3), group(1), pads(1,1), strides(1,1), dilations(1,1)
+        model = "models/conv_fp16.onnx" if self.args.precision == "fp16" else "models/conv_fp32.onnx"
+        data_type = np.float16 if self.args.precision == "fp16" else np.float32
+
+        # change here to test your data shape
+        self.add_case(OpParam(2, 320, 320, 64, 64, data_type), model)
+
+    def case_profile(cls, op_param, time):
```
Comment (mindest) [inline onnxruntime/python/tools/microbench/conv.py:46]: Same here.

**Comment by mindest** [comment]
Looks the decorator is also missing in cases for `cast`, `fast_gelu`, `matmul`, `skip_layer_norm`.

File: onnxruntime/python/tools/microbench/conv.py (hunk 1 above)
Comment (PeixuanZuo) [reply onnxruntime/python/tools/microbench/conv.py:29]: Done.

File: onnxruntime/python/tools/microbench/conv.py (hunk 2 above)
Comment (PeixuanZuo) [reply onnxruntime/python/tools/microbench/conv.py:48]: Done.

**Review by mindest — APPROVED** [review]

**Pull request merged**